### PR TITLE
Filter by color options should have radio buttons insted of Checkboxes

### PR DESCRIPTION
**Pull request by Darshan-upadhyay1110**
- Added radio buttons in  `filter by color` options which had Checkboxes
 - it should have radio buttons same as we have in LO core
 - it also makes sense because we only need to apply one filter at a time in cases like `filterByColor`

Steps to check patch

    - Open .ods file
    - add some dummy data in different rows, but column should be the same
    - click on the top cell of dummy data
    - got to `DATA` tab and apply `AutoFilter`
    - open auto filter popup and select filer by color
    - compare previous version with the current patch
    - check same thing in LO core
Signed-off-by: Darshan-upadhyay1110 <darshan.upadhyay@collabora.com>

Change-Id: I00b4d84e5117b3c1f05fbc37c61331d2c3bc2a31

### CODE REVIEW

File: browser/src/control/jsdialog/Widget.TreeView.js [hunk 1]
```diff
@@ -362,7 +379,7 @@ function _headerlistboxEntry(parentContainer, treeViewData, entry, builder) {
 			td.innerText = entry.columns[i].text;
 
 		if (!disabled)
-			$(td).click(clickFunction);
+			$(parentContainer).click(clickFunction);
```
Comment (eszkadev) [inline browser/src/control/jsdialog/Widget.TreeView.js:382]: why is that needed?

File: browser/src/control/jsdialog/Widget.TreeView.js [hunk 2]
```diff
@@ -331,14 +342,20 @@ function _headerlistboxEntry(parentContainer, treeViewData, entry, builder) {
 
 	if (entry.state !== undefined) {
 		var td = L.DomUtil.create('td', '', parentContainer);
-		var checkbox = _createCheckbox(td, treeViewData, builder, entry);
+		var toggleElement;
+		if (singleClick) {
```
Comment (eszkadev) [inline browser/src/control/jsdialog/Widget.TreeView.js:346]: here will be `data.checkboxtype === 'radio'` or something similar

File: browser/src/control/jsdialog/Widget.TreeView.js (hunk 1 above)
Comment (Darshan-upadhyay1110) [reply browser/src/control/jsdialog/Widget.TreeView.js:382]: td only covers the label part of the option. it does not fire the `clickfunction` if we do check/uncheck the radio button.

`parentContainer` will cover that case where if user click on label or on checkbox, it will fire the `clickFunction`

**Comment by Darshan-upadhyay1110** [comment]
CORE changes :  https://gerrit.libreoffice.org/c/core/+/161043

Before:
image

After:
image

**Comment by Darshan-upadhyay1110** [comment]
image

Tested cypress on local, passing all the cases

File: browser/src/control/jsdialog/Widget.TreeView.js [hunk 3]
```diff
@@ -394,6 +405,19 @@ function _hasIcon(columns) {
...
+function _createSelectionElement (parentContainer, treeViewData, entry, builder) {
+	var selectionElement;
+	switch (treeViewData.checkboxtype) {
+		case 'radio':
+			selectionElement = _createRadioButton(parentContainer, treeViewData, builder, entry);
+			break;
+		case 'checkbox':
+			selectionElement = _createCheckbox(parentContainer, treeViewData, builder, entry);
+			break;
+	}
```
Comment (eszkadev) [inline browser/src/control/jsdialog/Widget.TreeView.js:417]: Core could send empty checkboxtype.
can we do this like: if (checkboxtype === radio) createRadio else createCheckbox ?

Comment (Darshan-upadhyay1110) [reply browser/src/control/jsdialog/Widget.TreeView.js:417]: i am afraid we can't do if else because some cases does not have neither checkbox nor radio.

Comment (eszkadev) [reply browser/src/control/jsdialog/Widget.TreeView.js:417]: but this is done only when `entry.state` exists which means we need one of those

Comment (Darshan-upadhyay1110) [reply browser/src/control/jsdialog/Widget.TreeView.js:417]: @eszkadev yes you are right for those cases where no radio or checkbox the entry.state will be undefined and it will never call the _createSelectionElement

Comment (Darshan-upadhyay1110) [reply browser/src/control/jsdialog/Widget.TreeView.js:417]: Okay got it i am updating switch with if else

**Review by eszkadev — APPROVED** [review]

**Review by pedropintosilva — APPROVED** [review]
Thank you
